src/tests/ftest/server/metadata_svc_ops.py

```diff
@@ -0,0 +1,139 @@
+               time has passed.
+            5. Create pool2 with property svc_ops_enable:0.
+            6. Create containers by ThreadManager on pool2.
+            7. Repeat test step 4 on pool2.
+            8. Compare metadata workload test time with and without duplicate rpc detection.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium,md_on_ssd
+        :avocado: tags=server,metadata
+        :avocado: tags=DuplicateRpcDetection,test_metadata_dup_rpc
+        """
+        self.dmg.server_set_logmasks("DEBUG", raise_exception=False)
+        cont_num = self.params.get("number_thread", '/run/metadata/*', default=1)
+        w_cycles = self.params.get("workload_test_cycles", '/run/metadata/*', default=5000)
+        t_loops = self.params.get("test_loops", '/run/metadata/*', default=10)
+        threshold_factor = self.params.get("threshold_factor", '/run/metadata/*', default=1.75)
+
+        self.log_step("Create pool with properties svc_ops_entry_age.")
+        pool1 = self.get_pool(dmg=self.dmg.copy())
+
+        self.log_step("Create containers by ThreadManager.")
+        container_manager = ThreadManager(
+            self.metadata_workload_test, self.get_remaining_time() - 30)
+        container_manager.add(
+            pool=pool1, cont_num=cont_num, workload_cycles=w_cycles, test_loops=t_loops)
+
+        self.log_step("Run specified metadata workload cycles in multiple test loops.")
+        results = container_manager.run()
+        num_failed = len(list(filter(lambda r: not r.passed, results)))
+        if num_failed > 0:
+            self.fail('#{} container create threads failed'.format(num_failed))
+
+        self.log_step("Create pool2 with property svc_ops_enable:0.")
+        params = {}
+        params['properties'] = "svc_ops_enabled:0"
+        self.add_pool(**params)
+
+        self.log_step("Create containers by ThreadManager on pool2.")
+        container_manager = ThreadManager(
+            self.metadata_workload_test, self.get_remaining_time() - 30)
+        container_manager.add(
+            pool=self.pool, cont_num=cont_num, workload_cycles=w_cycles, test_loops=t_loops)
+
+        self.log_step("Run specified metadata workload cycles in multiple test loops on pool2.")
+        base_results = container_manager.run()
+        average_time = statistics.mean(base_results[0].result)
+        num_failed = len(list(filter(lambda r: not r.passed, base_results)))
+        if num_failed > 0:
+            self.fail('#{} container create threads failed'.format(num_failed))
+
+        self.log_step(
+            "Compare metadata workload test time with and without duplicate rpc detection.")
+        self.log.info("pool1 results = %s", results[0].result)
+        self.log.info("baseline results = %s", base_results[0].result)
+        self.log.info("average baseline result= %s", average_time)
+        for result in results[0].result:
```

Comment:
Test steps documentation for step 4 should not specify that it is calculating an average for those loops following svc_ops_entry_age time.

This verification seems OK, since we expect the "early" loops before svc_ops_entry_age to be quicker than what the performance will "stabilize" to after svc_ops_entry_age time. i.e., all timings before and after svc_ops_entry_age time should fit underneath the baseline times the threshold_factor. But we particularly care about the iterations after svc_ops_entry_age time has passed.

Reply:
Please see the new commit, which should have addressed all the comments.